### PR TITLE
plan: support aggregation function contains aggregation function

### LINKED ISSUE #4472: incorrect "Invalid use of group function" error
Please answer these questions before submitting your issue. Thanks!

1. What did you do?

`select sum(distinct(if('a', (select adddate(elt(999, count(*)), interval 1 day)), .1))) as foo;`

2. What did you expect to see?

```
mysql> select sum(distinct(if('a', (select adddate(elt(999, count(*)), interval 1 day)), .1))) as foo;
+------+
| foo  |
+------+
|  0.1 |
+------+
1 row in set, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+---------------------------------------+
| Level   | Code | Message                               |
+---------+------+---------------------------------------+
| Warning | 1292 | Truncated incorrect DOUBLE value: 'a' |
+---------+------+---------------------------------------+
1 row in set (0.00 sec)
```

3. What did you see instead?

```
mysql> select sum(distinct(if('a', (select adddate(elt(999, count(*)), interval 1 day)), .1))) as foo;
ERROR 1111 (HY000): Invalid use of group function
```

4. What version of TiDB are you using (`tidb-server -V`)?

```
mysql> select tidb_version();
+-----------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                          |
+-----------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: 0.9.0
Git Commit Hash: e44227333d0106b2a846aede28e0afdfe57aa69c
Git Branch: master
UTC Build Time: 2017-09-07 12:37:03 |
+-----------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by XuHuaiyu**
fix #4472 
PTAL @winkyao @hanfei1991

### CODE REVIEW

File: plan/validator.go [hunk 1]
```diff
@@ -46,11 +46,6 @@ type validator struct {
 func (v *validator) Enter(in ast.Node) (out ast.Node, skipChildren bool) {
 	switch node := in.(type) {
 	case *ast.AggregateFuncExpr:
-		if v.inAggregate {
```
Comment (winkyao) [inline plan/validator.go]: is it safe to remove these ?

Comment (XuHuaiyu) [reply plan/validator.go]: This is a grammar check, I'm not sure why we abandon this before.
@shenli PTAL

**Comment by XuHuaiyu** [comment]
PTAL @coocood

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**